lib/pending-processor.js

```diff
@@ -250,11 +270,14 @@ function applyPendingChange(pendingChange, tries, callback) {
   }
 }
 
-module.exports = function(syncStorageImpl, dataHandlersImpl, hashProviderImpl, metricsClientImpl) {
+module.exports = function(syncStorageImpl, dataHandlersImpl, hashProviderImpl, metricsClientImpl, pendingWorkerRetryLimit) {
   syncStorage = syncStorageImpl;
   dataHandlers = dataHandlersImpl;
   hashProvider = hashProviderImpl;
   metricsClient = metricsClientImpl;
+  if (_.isNumber(pendingWorkerRetryLimit) && pendingWorkerRetryLimit > 0) {
+    retryLimit = pendingWorkerRetryLimit
+  }
   return function(pendingChangeRequest, callback) {
     var pendingChange = pendingChangeRequest.payload;
     var tries = pendingChangeRequest.tries;
```

Comment:
This works fine but initially tries will be `undefined`. Then checks for update and collision work fine because
`undefined < retryLimit` => false


Adding 0 explicitly instead of undefined actually causing bug:
```
pendingChangeRequest.tries = pendingChangeRequest.tries || 0;
var tries = pendingChangeRequest.tries`
```
When tries is 0 
`0 < retryLimit` => true and this means that conflicts that come from successful apply will not be saved.

TL;DR this code works , but we could set initial tries explicitly.